flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -4480,18 +4559,39 @@ class ArrayExprLowering {
     std::string globalName = Fortran::lower::mangle::mangleArrayLiteral(x);
     fir::GlobalOp global = builder.getNamedGlobal(globalName);
     if (!global) {
-      global = builder.createGlobalConstant(
-          loc, arrTy, globalName,
-          [&](fir::FirOpBuilder &builder) {
-            Fortran::lower::StatementContext stmtCtx;
-            fir::ExtendedValue result =
-                Fortran::lower::createSomeInitializerExpression(
-                    loc, converter, toEvExpr(x), symMap, stmtCtx);
-            mlir::Value castTo =
-                builder.createConvert(loc, arrTy, fir::getBase(result));
-            builder.create<fir::HasValueOp>(loc, castTo);
-          },
-          builder.createInternalLinkage());
+      mlir::Type symTy = arrTy;
+      auto eleTy = symTy.cast<fir::SequenceType>().getEleTy();
+      /* If we have a rank-1 array of integer, real, or logical, then we can
+       * create a global array with the dense attribute.
+       *
+       * The mlir tensor type can only handle integer, real, or logical. It
+       * does not currently support nested structures which is required for
+       * complex.
+       *
+       * Also, we currently handle just rank-1 since tensor type assumes
+       * row major array ordering. We will need to reorder the dimensions
+       * in the tensor type to support Fortran's column major array ordering.
+       * How to create this tensor type is to be determined.
+       */
```

Comment:
I don't usually see this form of commenting (/* .... */).  Is there a standard usage for this?

Reply:
I will switch to // style comments.